pynocular/database_model.py

```diff
@@ -514,6 +598,38 @@ async def save(self) -> None:
             for field in self._db_managed_fields:
                 setattr(self, field, row[field])
 
+    def get_primary_id(self) -> Any:
+        """Standard interface for returning the id of a field
+
+        This assumes that there is a single primary id, otherwise this returns `None`
+
+        Returns:
+            The ID value for this DatabaseModel instance
+
+        """
+        if len(self._primary_keys) > 1:
+            return None
+
+        return getattr(self, self._primary_keys[0].name)
+
+    async def fetch(self, resolve_references: bool = False) -> None:
+        """Gets the latest of the object from the database and updates itself
+
+        Args:
+            resolve_references: If True, resolve any foreign key references
+
+        """
+        # Get the latest version of self
+        get_params = {
+            primary_key.name: getattr(self, primary_key.name)
+            for primary_key in self._primary_keys
+        }
+        get_params["resolve_references"] = resolve_references
+        new_self = await self.get(**get_params)
```

Comment:
You can just `await self.get(**get_params, resolve_references=resolve_references)`